raiden/transfer/channel.py

```diff
@@ -388,6 +405,154 @@ def is_valid_lockedtransfer(
     )
 
 
+def is_valid_lockexpired(
+        state_change: ReceiveLockExpired,
+        channel_state: NettingChannelState,
+        sender_state: NettingChannelEndState,
+        receiver_state: NettingChannelEndState,
+) -> MerkletreeOrError:
+    message_name = 'LockExpired'
+    received_balance_proof = state_change.balance_proof
+    lock = channel_state.partner_state.secrethashes_to_lockedlocks.get(state_change.secrethash)
+
+    if not lock:
+        # If we don't find the lock,
+        # it must have been removed earlier.
+        msg = 'Lock with secrethash {} does not exist'.format(state_change.secrethash)
+        return (False, msg, None)
+
+    current_balance_proof = get_current_balanceproof(sender_state)
+    merkletree = compute_merkletree_without(sender_state.merkletree, lock.lockhash)
+
+    _, _, current_transferred_amount, current_locked_amount = current_balance_proof
+    expected_nonce = get_next_nonce(sender_state)
+    expected_locked_amount = current_locked_amount - lock.amount
+    transferred_amount_after_unlock = (
+        received_balance_proof.transferred_amount +
+        expected_locked_amount
+    )
+
+    if get_status(channel_state) != CHANNEL_STATE_OPENED:
+        msg = 'Invalid {} message. The channel is already closed.'.format(message_name)
+        result = (False, msg, None)
+
+    elif merkletree is None:
+        msg = 'Invalid {} message. Same lockhash handled twice.'.format(message_name)
+        result = (False, msg, None)
+
+    else:
+        locksroot_without_lock = merkleroot(merkletree)
+
+        (is_valid, signature_msg) = is_valid_signature(
+            received_balance_proof,
+            sender_state.address,
+        )
+
+        if not is_valid:
+            # The signature must be valid, otherwise the balance proof cannot be
+            # used onchain
+            msg = 'Invalid {} message. {}'.format(message_name, signature_msg)
+
+            result = (False, msg, None)
+
+        elif received_balance_proof.chain_id != channel_state.chain_id:
+            msg = (
+                "Invalid {} message. "
```

Comment:
Question for this `msg` and all the ones below in this function. Why do you have `message_name` as a variable even though it's basically `LockExpired` everywhere? If you don't need to have it variable please just hardcode it.